controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -0,0 +1,387 @@
+			RoleARN:   rosaScope.ControlPlane.Spec.RolesRef.StorageARN,
+		},
+		{
+			Name:      "cloud-credentials",
+			Namespace: "openshift-cloud-network-config-controller",
+			RoleARN:   rosaScope.ControlPlane.Spec.RolesRef.NetworkARN,
+		},
+		{
+			Name:      "kube-controller-manager",
+			Namespace: "kube-system",
+			RoleARN:   rosaScope.ControlPlane.Spec.RolesRef.KubeCloudControllerARN,
+		},
+		{
+			Name:      "kms-provider",
+			Namespace: "kube-system",
+			RoleARN:   rosaScope.ControlPlane.Spec.RolesRef.KMSProviderARN,
+		},
+		{
+			Name:      "control-plane-operator",
+			Namespace: "kube-system",
+			RoleARN:   rosaScope.ControlPlane.Spec.RolesRef.ControlPlaneOperatorARN,
+		},
+		{
+			Name:      "capa-controller-manager",
+			Namespace: "kube-system",
+			RoleARN:   rosaScope.ControlPlane.Spec.RolesRef.NodePoolManagementARN,
+		},
+	} {
+		roles = append(roles, cmv1.NewOperatorIAMRole().
+			Name(role.Name).
+			Namespace(role.Namespace).
+			RoleARN(role.RoleARN))
+	}
+	stsBuilder = stsBuilder.OperatorIAMRoles(roles...)
+
+	instanceIAMRolesBuilder := cmv1.NewInstanceIAMRoles()
+	instanceIAMRolesBuilder.MasterRoleARN("TODO")
+	instanceIAMRolesBuilder.WorkerRoleARN("TODO")
+	stsBuilder = stsBuilder.InstanceIAMRoles(instanceIAMRolesBuilder)
+	stsBuilder.OidcConfig(cmv1.NewOidcConfig().ID(*rosaScope.ControlPlane.Spec.OIDCID))
+
+	stsBuilder.AutoMode(true)
+
+	awsBuilder := cmv1.NewAWS().
+		AccountID(*rosaScope.ControlPlane.Spec.AccountID)
+	awsBuilder = awsBuilder.SubnetIDs(rosaScope.ControlPlane.Spec.Subnets...)
+	awsBuilder = awsBuilder.STS(stsBuilder)
+	clusterBuilder = clusterBuilder.AWS(awsBuilder)
+
+	clusterNodesBuilder := cmv1.NewClusterNodes()
+	clusterNodesBuilder = clusterNodesBuilder.AvailabilityZones("us-west-2a")
+	clusterBuilder = clusterBuilder.Nodes(clusterNodesBuilder)
+
+	clusterProperties := map[string]string{}
+	clusterProperties["rosa_creator_arn"] = *rosaScope.ControlPlane.Spec.CreatorARN
+
+	clusterBuilder = clusterBuilder.Properties(clusterProperties)
+	clusterSpec, err := clusterBuilder.Build()
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to create description of cluster: %v", err)
+	}
+
+	// Create a logger that has the debug level enabled:
+	ocmLogger, err := sdk.NewGoLoggerBuilder().
+		Debug(true).
+		Build()
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to build logger: %w", err)
+	}
+
+	// Create the connection, and remember to close it:
+	token := os.Getenv("OCM_TOKEN")
+	connection, err := sdk.NewConnectionBuilder().
+		Logger(ocmLogger).
+		Tokens(token).
+		URL("https://api.stage.openshift.com").
+		Build()
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to build connection: %w", err)
+	}
+	defer connection.Close()
+
+	log := logger.FromContext(ctx)
+	cluster, err := connection.ClustersMgmt().V1().Clusters().
+		Add().
+		// Parameter("dryRun", *config.DryRun).
+		Body(clusterSpec).
+		Send()
```

Comment:
Is the `Send()` smart enough to differentiate between a create and an update?

Reply:
I don't think so :) This will just fail if the cluster has already been created.
This is working fine for now, but we will need to differentiate between create/update for sure.